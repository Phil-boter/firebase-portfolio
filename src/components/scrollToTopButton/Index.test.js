import React from "react";

import { shallow, render } from "enzyme";

import ScrollToTopButton from "./Index";

const showScroll = true;

describe("ScrollToTopButton component", () => {
    let wrapper = shallow(<ScrollToTopButton showScroll={showScroll} />);
    it("should render ScrollToTopButton correctly and match snapshot", () => {
        const { container } = render(<ScrollToTopButton />);
        expect(container).toMatchSnapshot();
    });

    it("should render the ScrollToTop Button container", () => {
        expect(wrapper.find(".scroll-top-button").length).toBe(1);
    });
    it("should not render the ScrollToTopButton when offset < 400", () => {
        let offWrapper = shallow(<ScrollToTopButton showScroll={false} />);
        expect(offWrapper.find(".scroll-top-button").length).toBe(0);
    });
    it("scrolls to position top:0 when button is clicked", () => {
        const scrollToSpy = jest.fn();
        global.scrollTo = scrollToSpy;
        wrapper.find(".scroll-top-button").simulate("click");
        expect(scrollToSpy).toHaveBeenCalled();
        expect(window.scrollTo).toBeCalledWith({ top: 0, behavior: "smooth" });
    });
});
