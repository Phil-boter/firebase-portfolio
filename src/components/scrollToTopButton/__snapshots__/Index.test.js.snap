// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTopButton component should render ScrollToTopButton correctly and match snapshot 1`] = `undefined`;
